sys/mips/ebase.S

```diff
@@ -395,11 +388,20 @@ user_exc_leave:
         ori     t0, SR_EXL
         sw      t0, PCPU_SR(s0)
 
-        # If user has FPU disabled, don't restore its registers.
-        ext     t1, t0, SR_CU1_SHIFT, 1
+        # If user FPU context is not saved, don't restore its registers.
+        LOAD_PCPU(t0)
+        lw      t0, PCPU_CURTHREAD(t0)
+        lw      t1, TD_PFLAGS(t0)
+        and     t1, t1, TDP_FPCTXSAVED
         beqz    t1, skip_fpu_restore
         nop
 
+        # Clear TD_FPCTXSAVED flag.
+        lw      t1, TD_PFLAGS(t0)
+        nor     t2, zero, TDP_FPCTXSAVED
+        and     t1, t1, t2
+        sw      t1, TD_PFLAGS(t0)
+
```

Comment:
This could be optimized a little bit:
```
        # If user FPU context is not saved, don't restore its registers.
        ext     t1, t0, SR_CU1_SHIFT, 1
        LOAD_PCPU(t0)
        lw      t0, PCPU_CURTHREAD(t0)
        lw      t1, TD_PFLAGS(t0)
        and     t2, t1, TDP_FPCTXSAVED
        beqz    t2, skip_fpu_restore
        nop


        # Clear TD_FPCTXSAVED flag.
        # We know that t2 == TDP_FPCTXSAVED and that the flag is set,
        # so we can just xor t2 with td_pflags.
        xor     t1, t1, t2
        sw      t1, TD_PFLAGS(t0)
```

Reply:
done - thanks